Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,170 +14,191 @@
+    return _maxNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalPower(NSDecimal* result, const NSDecimal* number, NSUInteger power, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)minimumDecimalNumber {
+    static dispatch_once_t __minNumberOnce;
+    dispatch_once(&__minNumberOnce, ^{
+        _minNumber = [[NSDecimalNumber alloc] initWithMantissa:std::numeric_limits<unsigned long long>::min() exponent:127 isNegative:YES];
+    });
+    return _minNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-void NSDecimalRound(NSDecimal* result, const NSDecimal* number, NSInteger scale, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
++ (NSDecimalNumber*)decimalNumberWithDecimal:(NSDecimal)decimal {
+    return [[[NSDecimalNumber alloc] initWithDecimal:decimal] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSString* NSDecimalString(const NSDecimal* dcm, id locale) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)decimalNumberWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)isNegative {
+    return [[[NSDecimalNumber alloc] initWithMantissa:mantissa exponent:exponent isNegative:isNegative] autorelease];
 }
 
 /**
  @Status Stub
  @Notes
 */
-NSCalculationError NSDecimalSubtract(NSDecimal* result,
-                                     const NSDecimal* leftOperand,
-                                     const NSDecimal* rightOperand,
-                                     NSRoundingMode roundingMode) {
++ (NSDecimalNumber*)decimalNumberWithString:(NSString*)numericString locale:(id)locale {
     UNIMPLEMENTED();
     return StubReturn();
 }
 
-@implementation NSDecimalNumber
-
 /**
- @Status Caveat
- @Notes does not support locale using ',', does not support exponents
+ @Status Interoperable
 */
-+ (NSDecimalNumber*)decimalNumberWithString:(NSString*)str {
-    const char* pStr = [str UTF8String];
-    if (strstr(pStr, ".") != NULL) {
-        double fVal = strtod(pStr, NULL);
-        return [[self alloc] initWithDouble:fVal];
-    } else {
-        int64_t val = _strtoi64(pStr, NULL, 10);
-        return [[self alloc] initWithLongLong:val];
+- (instancetype)initWithDecimal:(NSDecimal)decimal {
+    if (self = [super init]) {
+        _decimal = new NSDecimal{ 0 };
+        NSDecimal* foo = new NSDecimal{ 0 };
+        if (_decimal == NULL) {
+            [self release];
+            return nil;
+        }
+        NSDecimalCopy(_decimal, &decimal);
+        NSDecimalCompact(_decimal);
     }
+
+    return self;
 }
 
 /**
  @Status Interoperable
 */
-- (const char*)objCType {
-    return "d";
+- (instancetype)initWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)flag {
+    NSDecimal decimal = { 0 };
+    decimal._exponent = exponent;
+    decimal._isNegative = flag;
+
+    std::vector<unsigned short> digits;
```

Comment:
>std::vector<unsigned short> digits; [](start = 4, length = 35)

You don't need a separate vector for this. Store the digits directly into decimal._mantissa, then reverse them in place.